src/twisted/web/test/test_flatten.py

```diff
@@ -706,15 +706,15 @@ def render(self, request: Optional[IRequest]) -> Flattenable:
                     Exception while flattening:
                       \\[<unrenderable>\\]
                       <unrenderable>
-                      .*
+                      <Deferred at .* current result: <twisted.python.failure.Failure builtins.RuntimeError: example>>
                       File ".*", line \\d*, in _flattenTree
-                        element = await element
-                    RuntimeError: example
+                        element = await element.*
                     """
                 ),
                 flags=re.MULTILINE,
             ),
         )
+        self.assertTrue("RuntimeError: example" in str(failure.value))
```

Comment:
I don't think that this is ok.

The previous check already has `builtins.RuntimeError: example>>` which should be the same thing.

So I don't see any extra value / info / assertion here.

I am not 100% what is the expected output.

Reply:
Why, exactly, is this not OK?

What we are trying to test for here is that the exception type is relayed somewhere and we have not, in our custom traceback construction logic, accidentally elided or changed it.  The previous check does not have RuntimeError in it any more, it was removed, due to the traceback formatting changes.  Ostensibly we *could* check it with an individual assertion, sure, but is it work any additional effort to figure out exactly what precise logic in Python's traceback formatting changed to offset this slightly differently than it was before?  We're testing Twisted's traceback construction, not Python's traceback formatting.

Reply:
I guess as long as we're spending so long discussing this line, I could point out that the new assertion loses information, so it would be better to change it to use the unittest idiom for this type of assertion rather than `assertTrue` (which is almost always the wrong choice and gives no information upon failure)

```suggestion
        self.assertIn("RuntimeError: example", str(failure.value))
```

Reply:
The check is not ok as I think that the previous check already does the same thing.

The assertion can be left, it there is no extra coverage for this assertion.

Reply:
Read the diff more carefully.  The previous check does not check that any more.

Reply:
Specifically, from the diff view: 
![image](https://user-images.githubusercontent.com/716529/200090321-7a9cf979-a498-4700-8b46-144824a5b632.png)

Reply:
I agree that this should be assertIn vs assertTrue, updated.